lib/internal/modules/esm/hooks.js

```diff
@@ -160,6 +168,15 @@ class Hooks {
       load,
     } = pluckHooks(exports);
 
+    const instance = {
```

Comment:
I would use something like `uid(3)` (it looks like [crypto.randomUUID()](https://nodejs.org/api/crypto.html#cryptorandomuuidoptions) is the only thing already available in node, but that would be massively overkill) or perhaps the loader's url. Chatting with Geoffrey, he pointed out a loader can now be registered multiple times, so if the loader's url, it would need to have something else appended, like a generation id.

Reply:
Ooow! That's a good idea!

Reply:
What’s the value of a symbol as opposed to a string like `file:///path/to/loader.js?1`? At least the latter provides some useful information, such as for debugging, and then we don’t need to have different data structures internally and externally.

Reply:
A symbol is guaranteed to be unique, so then there is no concern for a generational id on the same url:

```
Symbol(url) ≠ Symbol(url)
```

Also, the label of a symbol is visible, so it still provides almost identical debugging value 🙂

Reply:
We could also return a handle, like:

```js
const handle = register(...)
handle.deregister()
```

Then the handle could potentially get other methods in the future. It could also contain the return value of `initialize` as a property, like `handle.initializationResult`.

Not sure I prefer this over other options, just adding it to the mix. We still need to deal with the problem that `register` currently returns the return value of `initialize`, so either we add this symbol into `initialize` for the user to return, and then somehow we get symbols across the thread barrier; or we change the return signature of `register`.